autoload/gitmessenger/popup.vim

```diff
@@ -89,21 +89,30 @@ let s:popup.window_size = funcref('s:popup__window_size')
 
 function! s:popup__floating_win_opts(width, height) dict abort
     let bottom_line = line('w0') + winheight(0) - 1
-    if self.opened_at[1] + a:height <= bottom_line
+    let rownr = self.opened_at[1]
+    let colnr = self.opened_at[2]
+    " if the line is wrapped, calculate the colnr and colnr again
+    if colnr > &columns
```

Comment:
I think `&columns` is not available to check wrap because window may be split vertically. For example, let's say two windows are open by splitting vertically. If their width are equal, width of each window is smaller than `&columns / 2`.

I think we need to consider virtual edit on setting `self.opened_at`. For column, `virtcol()` function would be available. But for row, I'm not sure how we can get it.

Reply:
It seems that vim doesn't have any options to get the size of that "subwin", doesn't it?

Reply:
Hmmmm, it's difficult to solve in consideration of multi split windows, there is not way to get the cursor's position relative to the whole shell window...

How about I just close this PR just now?

Reply:
Yeah... actually it's difficult, I think. We can know virtual column by `virtcol()`. But calculating virtual row from virtual column and window width would be not easy. For example, we need to consider `breakindent` option.

> How about I just close this PR and wait for better solution?

It's fine for me. If you're ok, could you file a new issue for this? Then we are able to track this issue. Small description and link to this PR would be sufficient.

Reply:
Sure